packages/template-retail-react-app/app/pages/checkout/partials/contact-info.jsx

```diff
@@ -60,10 +69,17 @@ const ContactInfo = () => {
     const submitForm = async (data) => {
         setError(null)
         try {
-            await login(data)
+            if (!data.password) {
```

Comment:
It seems we add extra logic here. Can you provide some explanation why we need to call `updateCustomerForBasket` ?

Reply:
it's actually the same as the old login method. There are 2 fields on the contact info section. You can add your email to the basket and continue as guest, or login as a registered user.

Reply:
short answer: it's for "continue as guest".